### PR TITLE
Replace with enum two strings that were in strings.xml but were used as

### LINKED ISSUE #2931: Remove the strings which have to be stored as constant from strings.xml
**Summary:** 

As @yarons pointed out at #2927 , some strings from strings.xml should be removed. They should have been defined as constants. Ie.

\<string name="review_c_violation_report_problem">copyRightViolation\</string>

They don't need to be translated, they are strings to be used for the code, not for the user read them. 

This task is quite beginner friendly though.

### PR DESCRIPTION
**Pull request by kateOmally**
constants. Remove strings from strings.xml Issue # 2931

**Description (required)**
Two strings in strings.xml were being used as constants. Therefore, did not need to be strings, much less translatable strings, so changed to an enum and removed strings from strings.xml.

Fixes #2931 Remove the strings which have to be stored as constant from strings.xml

What changes did you make and why?
defined an enum, ReviewController.DelteReason, and removed strings from strings.xml.

**Tests performed (required)**
ReviewHelperTest
DeleteHelperTest
Ran app through the IDE on an Android device running 8.0.0. Used the Review feature.

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.
Tested betaDebug on HTC U Ultra phone with API level 26.

**Screenshots showing what changed (optional - for UI changes)**
No UI changes.

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._
yep.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java [hunk 1]
```diff
@@ -152,12 +153,12 @@ public void askReasonAndExecute(Media media, Context context, String question, S
         String[] reasonList = {"Reason 1", "Reason 2", "Reason 3", "Reason 4"};
 
 
-        if (problem.equals("spam")) {
+        if (problem == ReviewController.DeleteReason.SPAM) {
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/delete/DeleteHelper.java]: Why don't use equals?

**Review by neslihanturan — CHANGES_REQUESTED** [review]

**Comment by kateOmally** [comment]
If you like, I will change it to .equals().   If there's a convention for
using .equals instead of ==, i must have missed it. Please point me to the
right resource.

Correct me if I'm wrong:  I thank that as an enum, Java can use use either
== or .equals(). They're the same except that == provides compile time type
checking which .equals doesn't.

**Comment by ashishkumar468** [comment]
@kateOmally There is no convention regarding the usage of == or .equals. Its just that,  == operators for reference comparison (address comparison) and .equals() method for content comparison. As far as I understand (and I may be wrong), in this scenario, we do not need address comparison as well right ?

**Comment by kateOmally** [comment]
For strings, yes, but for enums, I believe the .equals() method just calls
the == operator

https://www.geeksforgeeks.org/comparing-enum-members-java/

I don't think it's that important either way. I thought == was more clear
because  it makes clear that we're not using a String. But in this case,
there's no danger of ambiguity, so it can go either way. Do you want me to
change it?

**Comment by ashishkumar468** [comment]
@kateOmally Thank you for reminding me, for the case of enums,  .equals(), it simply defers to ==. Actually == would be better as it is null safe as well. Concluding from this, IMO, the current state is good and you need not make any changes.

**Review by ashishkumar468 — APPROVED** [review]

**Comment by neslihanturan** [comment]
After reading this https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals seems like there is several opinions on this. My concern was about reference comparing, but since you explained this is not the case and @ashishkumar468 approved it, this PR fine for me too. Thanks for your first time contribution @kateOmally :tada:

**Review by neslihanturan — APPROVED** [review]

**Pull request merged**